PhysicsTools/NanoAOD/python/triggerObjects_cff.py

```diff
@@ -216,24 +216,40 @@
   selections = selections2016
 )
 
-from PhysicsTools.PatUtils.L1ECALPrefiringWeightProducer_cff import prefiringweight
+from PhysicsTools.PatUtils.L1PrefiringWeightProducer_cff import prefiringweight
 #Next line will be updated once we get UL2016 maps
-run2_jme_2016.toModify( prefiringweight, DataEra = cms.string("2016BtoH"))
+(run2_muon_2016 & tracker_apv_vfp30_2016).toModify( prefiringweight, DataEraECAL = cms.string("2016BtoH"),  DataEraMuon = cms.string("2016preVFP"))
+(run2_muon_2016 & ~tracker_apv_vfp30_2016).toModify( prefiringweight, DataEraECAL = cms.string("2016BtoH"),  DataEraMuon = cms.string("2016postVFP"))
 #Next line is for UL2017 maps 
-run2_jme_2017.toModify( prefiringweight, DataEra = cms.string("UL2017BtoF"))
+run2_jme_2017.toModify( prefiringweight, DataEraECAL = cms.string("UL2017BtoF"), DataEraMuon = cms.string("20172018"))
+#Next line is for UL2018 maps 
+run2_muon_2018.toModify( prefiringweight, DataEraECAL = cms.string("None"), DataEraMuon = cms.string("20172018"))
+
```

Comment:
we should add something for the run2_nanoAOD_102Xv1
I made this comments on the earlier version
https://github.com/cms-sw/cmssw/pull/32728/files#r563596034
now the changes for the muons indicate that something should happen.

can you clarify ?

Reply:
If I understand the old discussion correctly, we have indeed have to add the prefiring weights to the nanoAOD for 2018. Should this line be modified for that? https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/nano_cff.py#L106

Reply:
for what I see we do not store the 2018 at all even for UL 
https://cms-nanoaod-integration.web.cern.ch/integration/32728/mc106Xul16_doc.html#L1PreFiringWeight
https://cms-nanoaod-integration.web.cern.ch/integration/32728/mc106Xul17_doc.html#L1PreFiringWeight
https://cms-nanoaod-integration.web.cern.ch/integration/32728/mc106Xul18_doc.html

we do have the pre-UL (run2_nanoAOD_94X2016 | run2_nanoAOD_94XMiniAODv1 | run2_nanoAOD_94XMiniAODv2) 
i.e. https://cms-nanoaod-integration.web.cern.ch/integration/32728/mc94Xv2_doc.html#L1PreFiringWeight
but again not for the 2018 (i.e., with modifier  run2_nanoAOD_102Xv1)

can you check locally what happens for the 2018? and do we really need it ?

Reply:
I have added 2018 here: https://github.com/cms-sw/cmssw/pull/33758/files#diff-9b898944f08810913ea5c1605c47bd9a6c99eb91fc20f491ba2767702f4a0184R260 and confirmed that the prefiring weights are now added to 2018 nanoAOD. 

We indeed need to add these weights as they are non-negligible for muons for that year.